src/components/form/fieldCreator.d.ts

```diff
@@ -53,15 +58,15 @@ export interface FieldProps<T, F, P extends FieldComponentProps<F, ExtraParams>,
     type?: string
     label: string
     defaultValue?: F
-    required: boolean
-    readOnly: boolean
+    required?: boolean
+    readOnly?: boolean
     autoComplete?: AutoFill
     validator?: Validator
     mapping?: PathMapping
     format?: Formatter<T, F>
     rawProperty?: string
     component: ComponentType<P>
-    extendedParams?: ExtraParams | ((i18n: I18nResolver) => ExtraParams)
+    extendedParams?: Prettify<Omit<P, keyof FieldComponentProps<F, ExtraParams>>> | ((i18n: I18nResolver) => Prettify<Omit<P, keyof FieldComponentProps<F, ExtraParams>>>)
```

Comment:
This is super hard to read, quite non-idiomatic IMHO. Is the added-value here (would you mind describing it?) really worth it?

Reply:
Malheureusement en l'état je ne vois pas comment faire plus simple. À part éventuellement en créant un type un alias mais ça reviendrait au même...
Pour faire simple c'est un type générique, qui pour une liste de propriété d'un composant "field", qui doivent donc étendre le type `FieldComponentProps`, on extrait les paramètres non-standards qui n'appartiennent pas au type commun `FieldComponentProps`.

Pour illustrer, dans une version simplifiée on a `type FieldComponentProps<F> = { name: string, value: F }`.
Pour un composant `const CustomField = (props: FieldComponentProps<string> & { customProp: string }) => ...`
Alors `extendedParams`aura le type `{ customProp: string }`

`Prettify<A>` c'est un helper maison qui permet juste de simplifier le type-hint l'IDE pour simplement afficher la version "calculée" `{ customProp: string }` sinon ça affiche le type tel qu'il apparait là...

Tu notera qu'il y a un union qui permet simplement de passer le même type d'argument mais sous forme de retour d'une fonction qui prend l'utilisation i18n de localisation en argument.

Bref, c'est compliqué parce que toute la logique de création des formulaire est une usine a gaz que je m'efforce de typer comme je peux en attendant de pouvoir refaire tout ça... :/